src/coreclr/nativeaot/Runtime/wasm/PalRedhawkWasm.cpp

```diff
@@ -39,4 +39,101 @@ void PalGetMaximumStackBounds_SingleThreadedWasm(void** ppStackLowOut, void** pp
     *ppStackLowOut = pStackLow;
     *ppStackHighOut = pStackHigh;
 }
+
+#ifdef TARGET_WASI
+// nop stubs
+
+int pthread_mutex_init(pthread_mutex_t * m, const pthread_mutexattr_t *__restrict a)
+{
+    return 0;
+}
+
+int pthread_mutexattr_init(pthread_mutexattr_t *a)
+{
+    return 0;
+}
+
+int pthread_mutexattr_settype(pthread_mutexattr_t *a, int type)
+{
+    return 0;
+}
+
+int pthread_mutex_destroy(pthread_mutex_t *mutex)
+{
+    return 0;
+}
+
+int pthread_mutexattr_destroy(pthread_mutexattr_t *)
+{
+    return 0;
+}
+
+int pthread_cond_init(pthread_cond_t *cond, const pthread_condattr_t *attr)
+{
+    return 0;
+}
+
+int pthread_condattr_init(pthread_condattr_t *attr)
+{
+    return 0;
+}
+
+int pthread_mutex_lock(pthread_mutex_t *)
+{
+    return 0;
+}
+
+int pthread_mutex_unlock(pthread_mutex_t *)
+{
+    return 0;
+}
+
+pthread_t pthread_self(void)
+{
+    return (pthread_t)0;
+}
+
+int pthread_equal(pthread_t, pthread_t)
+{
+    return 1; // only one thread
+}
+
+int pthread_attr_init(pthread_attr_t *)
+{
+    return 0;
+}
+
+int pthread_attr_destroy(pthread_attr_t *)
+{
+    return 0;
+}
+
+int pthread_condattr_destroy(pthread_condattr_t *)
+{
+    return 0;
+}
+
+int pthread_cond_broadcast(pthread_cond_t *)
+{
+    return 0;
+}
+
+int pthread_attr_setdetachstate(pthread_attr_t *, int)
+{
+    return 0;
+}
+
+#include <sys/mman.h>
```

Comment:
Why is the `#include` needed?

Reply:
Will remove, looks like I don't need it.